File: Scripts/results/generated_code_results/RN_gpt/Productivity/CourseTaskManagerApp/ClassroomScreen.js
import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity, ImageBackground, Button } from 'react-native';

const ClassroomScreen = () => {
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.dateText}>Today, 17 Sept</Text>
        <Button title="+ Add Class" onPress={() => {}} />
      </View>

      <Text style={styles.title}>Classroom</Text>

      <ImageBackground
        source={require('../assets/snack-icon.png')}
        style={styles.classImage}
      >
        {/* Additional Image details such as tags could be here */}
      </ImageBackground>

      <View style={styles.classDetails}>
        <Text style={styles.subject}>📘 Physics</Text>
        <Text>⚛️ Chapter 3: Force</Text>
        <Text>👤 Alex Jesus</Text>
        <Text>🧑‍🎓 45 Enrolled</Text>
      </View>

      <View style={styles.tabs}>
        <TouchableOpacity>
          <Text style={styles.activeTab}>Description</Text>
        </TouchableOpacity>
        <TouchableOpacity>
          <Text>Discussion</Text>
        </TouchableOpacity>
        <TouchableOpacity>
          <Text>Resources</Text>
        </TouchableOpacity>
      </View>

      <View style={styles.description}>
        <Text>About this class</Text>
        <Text>
          Force: The push or pull on an object with mass that causes it to change its velocity.
          Force is an external agent capable of changing the state of rest or motion of a particular body.
          It has a magnitude and a direction. The direction towards which the force is applied is known as the direction of the force and the application of force is in a point where force is applied.
        </Text>
      </View>

      <View style={styles.navigationBar}>
        <TouchableOpacity onPress={() => {}}>
          <Text>Home</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => {}}>
          <Text>Tasks</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => {}}>
          <Text style={styles.activeNav}>Classroom</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => {}}>
          <Text>Chat</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
    backgroundColor: 'white',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  dateText: {
    fontSize: 16,
    color: 'gray',
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    marginVertical: 10,
  },
  classImage: {
    height: 200,
    borderRadius: 10,
    overflow: 'hidden',
    marginVertical: 10,
  },
  classDetails: {
    marginVertical: 10,
  },
  subject: {
    fontWeight: 'bold',
    fontSize: 18,
  },
  tabs: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginTop: 20,
  },
  activeTab: {
    fontWeight: 'bold',
    borderBottomWidth: 2,
    borderBottomColor: 'black',
  },
  description: {
    marginVertical: 20,
  },
  navigationBar: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 10,
    borderTopWidth: 1,
    borderColor: '#ddd',
  },
  activeNav: {
    fontWeight: 'bold',
    borderBottomWidth: 2,
    borderBottomColor: 'black',
  },
});

export default ClassroomScreen;

File: Scripts/results/generated_code_results/Flutter/bak_code/Productivity/CourseTaskManagerApp/ClassroomScreen.dart
import 'package:flutter/material.dart';
class ClassroomScreen extends StatefulWidget {
@override
_ClassroomScreenState createState() => _ClassroomScreenState();
}
class _ClassroomScreenState extends State<ClassroomScreen> with SingleTickerProviderStateMixin {
late TabController _tabController;
@override
void initState() {
super.initState();
_tabController = TabController(length: 3, vsync: this);
}
@override
void dispose() {
_tabController.dispose();
super.dispose();
}
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text('Today, 17 Sept', style: TextStyle(fontSize: 14)),
Text('Classroom', style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold)),
],
),
actions: [
ElevatedButton.icon(
onPressed: () {
// Add class functionality
},
icon: Icon(Icons.add),
label: Text('Add Class'),
style: ElevatedButton.styleFrom(
backgroundColor: Colors.green,
),
),
],
),
body: Column(
children: [
Stack(
children: [
Image.asset('assets/images/template.png', fit: BoxFit.cover, height: 200),
Positioned(
top: 10,
left: 10,
child: Container(
padding: EdgeInsets.symmetric(horizontal: 8, vertical: 4),
decoration: BoxDecoration(
color: Colors.red,
borderRadius: BorderRadius.circular(4),
),
child: Text('LIVE', style: TextStyle(color: Colors.white)),
),
),
],
),
Padding(
padding: const EdgeInsets.all(16.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Row(
children: [
Icon(Icons.school, color: Colors.blue),
SizedBox(width: 8),
Text('Physics', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
],
),
Text('Chapter 3: Force', style: TextStyle(color: Colors.grey)),
SizedBox(height: 8),
Row(
children: [
Icon(Icons.person, color: Colors.grey),
SizedBox(width: 8),
Text('Alex Jesus'),
Spacer(),
Icon(Icons.people, color: Colors.grey),
SizedBox(width: 4),
Text('45 enrolled'),
],
),
],
),
),
TabBar(
controller: _tabController,
tabs: [
Tab(text: 'Description'),
Tab(text: 'Discussion'),
Tab(text: 'Resources'),
],
),
Expanded(
child: TabBarView(
controller: _tabController,
children: [
SingleChildScrollView(
padding: EdgeInsets.all(16),
child: Text(
'About this class\nForce: The push or pull on an object with mass that causes it to change its velocity.\n\nForce is a vector quantity capable of changing the state of rest or motion of a particular body.\nIt has a magnitude and a direction. The direction in which the force is applied is known as the direction of the force and the application of force is in a point where force is applied.',
),
),
Center(child: Text('Discussion Tab Content')),
Center(child: Text('Resources Tab Content')),
],
),
),
],
),
bottomNavigationBar: BottomNavigationBar(
currentIndex: 2,
onTap: (index) {
if (index == 0) {
Navigator.pushNamed(context, 'Home');
} else if (index == 3) {
Navigator.pushNamed(context, 'Discussion');
}
},
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),
BottomNavigationBarItem(icon: Icon(Icons.list), label: 'Tasks'),
BottomNavigationBarItem(icon: Icon(Icons.school), label: 'Classroom'),
BottomNavigationBarItem(icon: Icon(Icons.chat), label: 'Chat'),
],
),
);
}
}